casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala

```diff
@@ -83,11 +83,12 @@ object BlockAPI {
     )
 
   def getListeningNameDataResponse[F[_]: Sync: MultiParentCasperRef: Log: SafetyOracle: BlockStore](
+      depth: Long,
       listeningName: Channel
   ): F[ListeningNameDataResponse] = {
     def casperResponse(implicit casper: MultiParentCasper[F], channelCodec: Codec[Channel]) =
       for {
-        mainChain           <- getMainChainFromTip[F]
+        mainChain           <- getMainChainFromTip[F](depth)
```

Comment:
Why do we only look at the main chain here anyways? Names could be effected anywhere in the DAG, so we should traverse the whole thing, not just a single chain.

Reply:
The main chain block's state hash will allow you to reset the tuplespace to everything you care about, and in any case, the other blocks will only get finalized once the main chain blocks are finalized.

Reply:
I see. So the idea is that a block with multiple parents implicitly also contains the state changes that occurred in its non-main parent?

Reply:
Indeed